storage/key_index.go

```diff
@@ -132,6 +132,30 @@ func (ki *keyIndex) get(atRev int64) (modified, created revision, ver int64, err
 	return revision{}, revision{}, 0, ErrRevisionNotFound
 }
 
+// since returns recorded valid revisions since the given rev. If multiple revisions
```

Comment:
If multiple revisions have the same main, it only returns the largest one. Because only the last event on one key in one transaction should be seen by caller.